silence/silence.go

```diff
@@ -225,6 +273,9 @@ Loop:
 // GC runs a garbage collection that removes silences that have ended longer
 // than the configured retention time ago.
 func (s *Silences) GC() (int, error) {
+	start := time.Now()
+	defer func() { s.metrics.gcDuration.Observe(time.Since(start).Seconds()) }()
```

Comment:
Maybe I'm answering myself here, my guess is that defer makes it clearer but the performance is more important in the Query method. (I saw that you did the same defer statement in the other maintenance methods)